dist/cms.js

```diff
@@ -77,11 +77,86 @@ function getWordPressScripts() {
   return entries;
 }
 
+/**
+ * Detects the type of WordPress content for the current document.
+ *
+ * @returns {object} Object with fields `contentType`, `postType`, and `taxonomy`.
+ */
+function getWordPressContentType() {
+  const content = {
+    contentType: 'unknown',
+    postType: '',
+    taxonomy: '',
+  };
+  try {
+    const bodyClass = document.body.classList;
+
+    if ( bodyClass.contains( 'home' ) ) {
+      /*
+       * The home page, either containing the blog,
+       * or a "static front page".
+       */
+      if ( bodyClass.contains( 'blog' ) ) {
+        content.contentType = 'home-blog';
+        content.postType = 'post';
+      } else if ( bodyClass.contains( 'page' ) ) {
+        content.contentType = 'home-page';
+        content.postType = 'page';
+      }
+    } else if ( bodyClass.contains( 'blog' ) ) {
+      /*
+       * The blog, separate from the home page.
+       * Only relevant if the home page contains a "static front page".
+       */
+      content.contentType = 'blog';
+      content.postType = 'post';
+    } else if ( bodyClass.contains( 'singular' ) ) {
+      /*
+       * Any singular content (other than the "static front page").
+       * Either a page, or content of another post type.
+       */
+      content.contentType = 'singular';
+      if ( bodyClass.contains( 'page' ) ) {
+        content.postType = 'page';
+      } else if ( bodyClass.contains( 'single' ) ) {
+        const postTypeClass = Array.from( bodyClass ).find( c => c.startsWith( 'single-' ) && ! c.startsWith( 'single-format-' ) );
+        if ( postTypeClass ) {
+          content.postType = postTypeClass.replace( 'single-', '' );
+        }
+      }
+    } else if ( bodyClass.contains( 'archive' ) ) {
+      /*
+       * Any archive (other than the blog).
+       * Either a category archive, a tag archive, a custom post type archive,
+       * or a custom taxonomy archive.
+       */
```

Comment:
What about `author` and `date` archives? Also `search`, but unlikely to be in HTTP Archive.

Reply:
The current approach uses the agnostic `archive` name, which would apply to any of them. Essentially, all that this does is that for post type or taxonomy archives, it also stores related metadata (post type / taxonomy).

I'm not sure there's value in storing additional data for those kinds of archives specifically, as I assume they're rarely (if ever) part of HTTP Archive data, as it's also common to discourage indexing those kinds of archives (e.g. in Yoast SEO).